debezium-server-eventhubs/src/main/java/io/debezium/server/eventhubs/BatchManager.java

```diff
@@ -187,4 +197,54 @@ private void emitBatchToEventHub(EventDataBatchProxy batch) {
             }
         }
     }
+
+    /**
+     * Applies the specified hash function to the input string.
+     *
+     * @param input the string to hash
+     * @param hashFunction the hash function to use
+     * @return the hashed string
+     */
+    private String applyHashFunction(String input, HashFunction hashFunction) {
+        switch (hashFunction) {
+            case JAVA:
+                return String.valueOf(input.hashCode());
+            case MD5:
+                return computeDigest(input, "MD5");
+            case SHA1:
+                return computeDigest(input, "SHA-1");
+            case SHA256:
+                return computeDigest(input, "SHA-256");
+            default:
+                throw new IllegalArgumentException("Unsupported hash function: " + hashFunction);
+        }
+    }
+
+    /**
+     * Computes a message digest hash for the input string.
+     *
+     * @param input the string to hash
+     * @param algorithm the digest algorithm (MD5, SHA-1, SHA-256)
+     * @return the hex-encoded hash string
+     */
+    private String computeDigest(String input, String algorithm) {
+        try {
+            MessageDigest digest = MessageDigest.getInstance(algorithm);
```

Comment:
Digest should be initialized only once. Is it needed to reinitialize it for every message?